DigitalLearningSolutions.Web.Tests/Controllers/LearningMenu/LearningMenuControllerTests.cs

```diff
@@ -583,6 +570,201 @@ public void Tutorials_should_return_404_if_unable_to_enrol()
                 .WithRouteValue("code", 404);
         }
 
+        [Test]
+        public void ContentViewer_should_StartOrUpdate_course_sessions_if_valid_tutorial()
+        {
+            // Given
+            var expectedTutorialContent = TutorialContentHelper.CreateDefaultTutorialContent();
+            const int progressId = 3;
+
+            A.CallTo(() => tutorialContentService.GetTutorialContent(CustomisationId, SectionId, TutorialId))
+                .Returns(expectedTutorialContent);
+            A.CallTo(() => courseContentService.GetOrCreateProgressId(CandidateId, CustomisationId, CentreId))
+                .Returns(progressId);
+
+            // When
+            controller.ContentViewer(CustomisationId, SectionId, TutorialId);
+
+            // Then
+            A.CallTo(() => sessionService.StartOrUpdateSession(CandidateId, CustomisationId, httpContextSession))
+                .MustHaveHappenedOnceExactly();
+            A.CallTo(() => sessionService.StartOrUpdateSession(A<int>._, A<int>._, A<ISession>._))
+                .WhenArgumentsMatch((int candidateId, int customisationId, ISession session) =>
+                    candidateId != CandidateId || customisationId != CustomisationId)
+                .MustNotHaveHappened();
+        }
+
+        [Test]
+        public void ContentViewer_should_not_StartOrUpdate_course_sessions_if_invalid_tutorial()
+        {
+            // Given
+            const int progressId = 3;
+
+            A.CallTo(() => tutorialContentService.GetTutorialContent(CustomisationId, SectionId, TutorialId))
+                .Returns(null);
+            A.CallTo(() => courseContentService.GetOrCreateProgressId(CandidateId, CustomisationId, CentreId))
+                .Returns(progressId);
+
+            // When
+            controller.ContentViewer(CustomisationId, SectionId, TutorialId);
+
+            // Then
+            A.CallTo(() => sessionService.StartOrUpdateSession(A<int>._, A<int>._, A<ISession>._))
+                .MustNotHaveHappened();
+        }
+
+        [Test]
+        public void ContentViewer_should_not_StartOrUpdate_course_sessions_if_unable_to_enrol()
+        {
+            // Given
+            var expectedTutorialContent = TutorialContentHelper.CreateDefaultTutorialContent();
+
+            A.CallTo(() => tutorialContentService.GetTutorialContent(CustomisationId, SectionId, TutorialId))
+                .Returns(expectedTutorialContent);
+            A.CallTo(() => courseContentService.GetOrCreateProgressId(CandidateId, CustomisationId, CentreId))
+                .Returns(null);
+
+            // When
+            controller.ContentViewer(CustomisationId, SectionId, TutorialId);
+
+            // Then
+            A.CallTo(() => sessionService.StartOrUpdateSession(A<int>._, A<int>._, A<ISession>._))
+                .MustNotHaveHappened();
+        }
+
+        [Test]
+        public void ContentViewer_should_UpdateProgress_course_sessions_if_valid_tutorial()

```

Comment:
`course_sessions` doesn't really make sense in this test name? 🙂

Reply:
Yep, renamed in b073a6a 🙂